pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1465,6 +1469,72 @@ public void uploadToDeepStoreIfMissing(TableConfig tableConfig, List<SegmentZKMe
     }
   }
 
+  public long deleteTmpSegments(String tableNameWithType, List<SegmentZKMetadata> segmentsZKMetadata) {
+    Preconditions.checkState(!_isStopping, "Segment manager is stopping");
+
+    if (!TableNameBuilder.isRealtimeTableResource(tableNameWithType)) {
+      return 0L;
+    }
+
+    TableConfig tableConfig = _helixResourceManager.getTableConfig(tableNameWithType);
+    if (tableConfig == null) {
```

Comment:
If table config can not be found, should we delete the tmp segments straight away? why skip?

Reply:
if (!isLowLevelConsumer(tableNameWithType, tableConfig) requires non null tableConfig

Reply:
to clean up orphan segments for deleted table, should have a different job since not only .tmp. but other files need to be deleted.